app/src/test/java/com/github/sdp_begreen/begreen/models/PostTest.kt

```diff
@@ -0,0 +1,24 @@
+package com.github.sdp_begreen.begreen.models
+
+import android.graphics.Bitmap
+import org.junit.Test
+import org.hamcrest.CoreMatchers.*
+import org.hamcrest.MatcherAssert.assertThat
+import org.mockito.Mockito.mock
+
+class PostTest {
+
+    private val testBitmap = mock(Bitmap::class.java)
```

Comment:
I am not sure that you should mock the `Bitmap` class here.

Mock testing are used to isolate the class under test by injecting fake dependency (mocked dependency) into this class, so that we don't depends on external implementation at test time.

An example would be if one of your class connect to the internet through an `HttpClient` interface, then in unit test you don't want to actually use the real implementation of the `HttpClient` you would rather `mock` the `HttpClient` to return fake response only for test time, to isolate only the logic of the class you are currently testing, not the behaviour of the `HttpClient` implementation.

For a better explanation, maybe you can have a look to the [test lecture](https://github.com/sweng-epfl/public/tree/main/lectures/Testing#how-can-one-test-entire-modules) from last semester that we had in Sweng.

But here, the way you are using it you are not actually mocking the `Bitmap` class, you are indeed creating a mock instance of `Bitmap` but you don't inject ii aywhere and you don't explicitely mock methods of the `Bitmap` class.

Here I think you simply need to create an fake `Bitmap` instance just to store it in the `Post` instance that you are testing.

Let me know if you have any questions about what I said.

Reply:
Thank you for the explanations. I've changed the mock with :

`private val testBitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888)`

This forced me to move the test to the androidTest folder otherwise the Bitmap is not recognized. I saw that somebody else had the same issue with the _UserTest_ and did the same.